deps/Makefile

```diff
@@ -79,7 +79,7 @@ ifeq ($(uname_S),SunOS)
 	LUA_CFLAGS= -D__C99FEATURES__=1
 endif
 
-LUA_CFLAGS+= -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC='' -DLUA_USE_MKSTEMP $(CFLAGS)
+LUA_CFLAGS+= -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DSERVER_STATIC='' -DLUA_USE_MKSTEMP $(CFLAGS)
```

Comment:
Not sure if this is even needed anymore.

Reply:
I agree, the PR that introduced it is 10 years old and we use static in a bunch of other places in the code base. Let's use this as an opportunity to simplify.